DomainContent/AvatarStore/attachmentItemScript.js

```diff
@@ -0,0 +1,156 @@
+            // attempt to touch the userData
+            var userData = JSON.parse(entityProperties.userData);
+            touchCallback.call(this, userData);
+            entityProperties.userData = JSON.stringify(userData);
+        } catch (e) {
+            print('Something went wrong while trying to touch/modify the userData. Could be invalid JSON or problem with the callback function.');
+        }
+    };
+
+    /**
+     * This callback is displayed as a global member.
+     * @callback touchJSONUserDataCallback
+     * @param {Object} userData
+     */
+
+    function AttachableItem() {
+        _this = this;
+    }
+    AttachableItem.prototype = {
+        preload : function(entityID) {
+            _entityID = entityID;
+            var properties = Entities.getEntityProperties(entityID);
+
+            _attachmentData = JSON.parse(properties.userData).Attachment;
+            if (_attachmentData.joint.indexOf(LEFT_RIGHT_PLACEHOLDER) !== -1) {
+                var baseJoint = _attachmentData.joint.substring(4);
+                _supportedJoints.push("Left".concat(baseJoint));
+                _supportedJoints.push("Right".concat(baseJoint));
+            } else {
+                _supportedJoints.push(_attachmentData.joint);
+            }
+
+            isAttached = _attachmentData.attached;
+
+            if (Entities.getNestableType(properties.parentID) !== "avatar" && !isAttached) {
+                messageChannel = MESSAGE_CHANNEL_BASE + properties.parentID;
+                Messages.subscribe(messageChannel);
+            }
+        },
+        startNearGrab: function(entityID, args) {
+            if (firstGrab) {
+                var properties = Entities.getEntityProperties(entityID);  
+                if (!properties.visible) {
+                    Entities.editEntity(entityID, {visible: true});
+                } 
+                firstGrab = false;
+            }
+        },
+            
+        releaseGrab: function(entityID, args) {
+            var hand = args[0] === "left" ? 0 : 1;
+            var properties = Entities.getEntityProperties(entityID);
+
+            if (Entities.getNestableType(properties.parentID) === "entity") {
+                Messages.sendMessage(messageChannel, "Removed Item :" + entityID);
+                Messages.unsubscribe(messageChannel); 
+                Entities.editEntity(entityID, {parentID: EMPTY_PARENT_ID});
+            }
+
+            var userData = properties.userData;
+            var position = properties.position; 
+            var attachmentData = JSON.parse(userData).Attachment;
+            isAttached = attachmentData.attached;
+
+            if (!isAttached) {
+                _supportedJoints.forEach(function(joint) {
+                    var jointPosition = MyAvatar.getJointPosition(joint);
+                    if (Vec3.distance(position, jointPosition) <= ATTACH_DISTANCE) {
+                        var newEntityProperties = Entities.getEntityProperties(_entityID);
```

Comment:
please filter on retrieving the userData `Entities.getEntityProperties(_entityID, 'userData');` because that`s the only property you're using.